### ISSUE TITLE
Ploting issue

### ISSUE THREAD

**Issue by kkkouadio** [AMMA-2050/metrics_workshop#53]
Hello,
We are computing the Monthly  Precipitation Extreme and got the attached error during the plotting step.
this is an example made for JJAS. 
generally we are able to compute but it seems difficult to make the plots using the platform. 

Best Regards!!!!!

Kouadio and Bamba

[calc_index_Monthly_precipitation_Extreme.py.gz](https://github.com/AMMA-2050/metrics_workshop/files/732981/calc_index_Monthly_precipitation_Extreme.py.gz)

[error.txt](https://github.com/AMMA-2050/metrics_workshop/files/732973/error.txt)

[index_Monthly_precipitation_Extreme_0.5x0.5_historical_jjas_West_Africa_MRI-ESM1.nc.gz](https://github.com/AMMA-2050/metrics_workshop/files/732968/index_Monthly_precipitation_Extreme_0.5x0.5_historical_jjas_West_Africa_MRI-ESM1.nc.gz)
[index_Monthly_precipitation_Extreme_0.5x0.5_historical_jjas_West_Africa_NorESM1-M.nc.gz](https://github.com/AMMA-2050/metrics_workshop/files/732967/index_Monthly_precipitation_Extreme_0.5x0.5_historical_jjas_West_Africa_NorESM1-M.nc.gz)

**Comment by cornkle** [comment]
Hi guys,
from what the error looks like your range_finder is already a 2D array. At the moment, the plot assumes it gets several time steps and just takes the first one: [0 , :, : ] 
Please try 
`cd = plt.contourf(lon,lat,range_finder, levs, cmap = plt.get_cmap('spectral'))`
so basically removing the  [0, :, :]  bracket that indicates a 3D array.

**Comment by kkkouadio** [comment]
Hello,
Thank you very much for reply. I am going to work on it again.
I also have some additional questions:
I am computed some statistics on let say PE_month
I used PE_month.data and got only the value and did some calculations without changing the shape of the cube. My question is how do I get back the metadata?
It is also seemed that one cannot use iris again when compute a cube with numpy. If true how to solve this? 
( when I try I got this error message: AttributeError: 'numpy.ndarray' object has no attribute 'aggregated_by')
See some lines below for more information.
Cheers.
Kouakou



perc_clim_data = np.repeat(perc_clim_data.reshape(perc_clim_data.shape[0],perc_clim_data.shape[1], perc_clim_data.shape[2]), cube_month.shape[0]/12, axis=0)
PE_month_att = cube_month - perc_clim_data
PE_month = PE_month_att.data
PE_month = np.where(PE_month >= 0, 1, 0)

print(PE_month)

PE_clim = PE_month.aggregated_by('month', iris.analysis.SUM)
 

 

    Le Vendredi 27 janvier 2017 11h46, cornkle <notifications@github.com> a écrit :
 

 Hi guys,
from what the error looks like your range_finder is already a 2D array. At the moment, the plot assumes it gets several time steps and just takes the first one: [0 , :, : ]
Please try
cd = plt.contourf(lon,lat,range_finder, levs, cmap = plt.get_cmap('spectral'))
so basically removing the [0, :, :] bracket that indicates a 3D array.—

**Comment by kkkouadio** [comment]
Hello,
We still working on the metric.  We were able to plot the percentile (just a step on the metric). However it still have some issues on the title and the labelbar.The size of the title did not change on the save figure (.eps) overthought it change on the shown one(.pdf). Don't know why.  Cheers
Kouadio & Bamba


 

    Le Vendredi 27 janvier 2017 11h46, cornkle <notifications@github.com> a écrit :
 

 Hi guys,
from what the error looks like your range_finder is already a 2D array. At the moment, the plot assumes it gets several time steps and just takes the first one: [0 , :, : ]
Please try
cd = plt.contourf(lon,lat,range_finder, levs, cmap = plt.get_cmap('spectral'))
so basically removing the [0, :, :] bracket that indicates a 3D array.—
